src/System.Text.Encoding/tests/NegativeEncodingTests.cs

```diff
@@ -39,18 +39,48 @@ public static unsafe void GetByteCount_Invalid(Encoding encoding)
         {
             // Chars is null
             Assert.Throws<ArgumentNullException>(encoding is ASCIIEncoding ? "chars" : "s", () => encoding.GetByteCount((string)null));
+#if netcoreapp
```

Comment:
in general I don't like to have many ifdef's in the code. we can wither move the cases that is netcoreapp specific to its own file or we can have test cases skipped for all platforms but netcoreapp. you may look at the PR https://github.com/dotnet/corefx/pull/16706 for similar idea.